zebra/zebra_dplane.h

```diff
@@ -1078,6 +1090,15 @@ void dplane_set_in_queue_limit(uint32_t limit, bool set);
 /* Retrieve the current queue depth of incoming, unprocessed updates */
 uint32_t dplane_get_in_queue_len(void);
 
+void dplane_ctx_set_vlan_ifindex(struct zebra_dplane_ctx *ctx,
+				 ifindex_t ifindex);
+ifindex_t dplane_ctx_get_vlan_ifindex(struct zebra_dplane_ctx *ctx);
+struct zebra_vxlan_vlan_array;
+void dplane_ctx_set_vxlan_vlan_array(struct zebra_dplane_ctx *ctx,
```

Comment:
I think we'd better have a comment that the ctx takes over the memory - it's not copying, right?

Reply:
we just do **ctx->u.vlan_info.vlan_array = vlan_array;** and free it when ctx is freed in dplane_ctx_free_internal().

Reply:
that's what I mean, exactly: the caller must allocate some memory, and must not free it, and the caller must not hand in info on the stack.

Reply:
I have added it in zebra_dplane.c  where the function is defined. 
https://github.com/FRRouting/frr/pull/16737/files#diff-8351b0eeaf9e92cf06ff9d3449825c62c2332dbe83109fe2df56d5606b5f75f5R3357

Reply:
hmm, gosh - I really tried to say exactly what I meant to say. I'll say more:
I want the dplane api to be better, and more useful. That includes being easier, safer to work with.
there are several patterns in the dplane code that involve structs or chunks of memory. there's a pattern where the dplane ctx has an embedded, fixed struct, and the caller's data is copied in. there's a pattern where the dplane ctx allocates memory, and copies the caller's data; the ctx allocates the memory and must free it. and then there's the worst possible pattern: the caller allocates some memory, and hands it over to the dplane ctx, who becomes responsible for it. that's particularly painful with our memory library, because the caller and the dplane ctx code must agree about the MTYPE used for the allocation. and that third pattern is the one you've chosen to use. I didn't reject that out of hand - though I was tempted to. but I really do want there to be a comment there in the header file so that if someone ever wants to know this crucial piece of information without actually reading the dplane source code, there will be some chance that the information will be highlighted, and the pattern will be reinforced that useful comments about critical behavior in the api are good - and even expected.

Reply:
Thanks, Mark for the detailed explanation. 
I will keep this in mind for future coding as well..